### DISCUSSION TITLE
Forkphorus in the editor - does that make sense?

### DISCUSSION THREAD

**Discussion by NoobTracker** [ScratchAddons/ScratchAddons#168] [Discussion]
So for example here (#166) there is the idea of having one of those compilers in the Scratch editor. Now here is why I'm really not sure if that makes sense:
When you change anything and then start the project, it has to save it, then refresh the forkphorus iframe and then start that and that's weird. Otherwise nothing is going to change. Now I think that it isn't the worst thing in the world to wait a second for fork to compile it, but saving usually takes at least 1 second and then refreshing also takes a few seconds. Then fork in its iframe has to download the script again and in the end it takes 20s until you can finally test your code. Yay.
So I wonder if that makes any sense at all to do that. Doing that for the main project kinda makes sense since you still want to see the instructions and stuff but in the editor? I'm not sure. What do you think?

Also, is there a way of not doing the reload stuff and just compiling it _somehow_?

**Comment by retronbv** [comment]
maybe have other player not in editor but on project page?

**Comment by NoobTracker** [comment]
> 
> 
> So I wonder if that makes any sense at all to do that. Doing that for the main project kinda makes sense since you still want to see the instructions and stuff but in the editor? 

I already mentioned that.

**Comment by retronbv** [comment]
oh oof

**Comment by NoobTracker** [comment]
Oh some good news: Currently they're trying to use cloud vars in fork.

**Comment by retronbv** [comment]
yesss

**Comment by retronbv** [comment]
but wait wont that mean if people use the fork orr turbo player they cant play with other peeps? unless thoose other peeps use too

**Comment by NoobTracker** [comment]
No, I guess it could access the same stuff Scratch also does. But I guess you can just try it by playing Othello or something in fork or just making a quick project with one button that just increments a cloud variable and then click that a few times in fork and see.

**Comment by NoobTracker** [comment]
Um I'm currently a new Scratcher since I'm banned and stuff so can anyone else test that here? https://forkphorus.github.io/?cloud=ws#409593079

**Comment by retronbv** [comment]
:o u banned?

**Comment by retronbv** [comment]
I can try real quick bc I’m supposed to be in a class rn

**Comment by retronbv** [comment]
I’m only scratcher

**Comment by retronbv** [comment]
U don’t need a account for fork

**Comment by NoobTracker** [comment]
I'm a New Scratcher. That's worse.
You still need to be signed in I think.

**Comment by retronbv** [comment]
Nope not for fork

**Comment by retronbv** [comment]
Just try

**Comment by NoobTracker** [comment]
And yeah I'm banned because someone misunderstood me and stuff.

**Comment by retronbv** [comment]
Banned on fork?

**Comment by NoobTracker** [comment]
Okay you're right you have to be on fork because that's different stuff

**Comment by NoobTracker** [comment]
no not on fork

**Comment by retronbv** [comment]
I’m in with someone

**Comment by retronbv** [comment]
Gtg

**Comment by NoobTracker** [comment]
That was me ...

**Comment by retronbv** [comment]
xD

**Comment by NoobTracker** [comment]
Okay you are right cloud vars are not what I expected them to be ...

**Comment by retronbv** [comment]
Lol

**Comment by retronbv** [comment]
Ask Garbo bc he made fork

**Comment by Explosion-Scratch** [comment]
The link you gave dosen't work with cloud :(

**Comment by retronbv** [comment]
Yes it does, we played

**Comment by Explosion-Scratch** [comment]
Hmm. Also, even if editing the project and embedding the phosphorus player
worked fine and all, wouldn't this raise issues with real time editing? If
you made an edit would it automatically instantly update?

**Comment by retronbv** [comment]
Yeah

**Comment by NoobTracker** [comment]
I already thought about that and I think that it would just change when you click the green flag - but since I personally really don't like real time editing, I don't care. Maybe that could even be possible somehow by inserting and moving code when you change something - dunno. You should talk to the creators of fork. 
And well, it would just be an option, not necessarily the default.

**Comment by retronbv** [comment]
@GarboMuffin ^

**Comment by WorldLanguages** [comment]
Using 3rd party editors while editing is basically impossible to achieve correctly.

**Comment by UltraCupcake** [comment]
But what if it works like turbowarp in editor but better: doesnt do the compiling -unless you press the flag AFTER you edited blocks (edit blocks again and it should stop the compiler)- when playing other people's projects, compile... that kind of stuff
